async-nats/src/lib.rs

```diff
@@ -903,8 +907,50 @@ impl Subscriber {
     ///  subscriber.unsubscribe();
     /// # Ok(())
     /// # }
-    pub fn unsubscribe(self) {
-        drop(self)
+    pub async fn unsubscribe(&mut self) {
+        self.receiver.close();
```

Comment:
Probably not immediately close, but instead let buffered things come through.

Reply:
that should be the current behaviour according to `close` docs:
```
    /// Closes the receiving half of a channel without dropping it.
    ///
    /// This prevents any further messages from being sent on the channel while
    /// still enabling the receiver to drain messages that are buffered. Any
    /// outstanding [`Permit`] values will still be able to send messages.
    ///
```